src/main/scala/rps/Main.scala

```diff
@@ -5,26 +5,36 @@ import scala.io.StdIn
 import akka.actor.ActorSystem
 import akka.stream.ActorMaterializer
 import akka.http.scaladsl.Http
-import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server._
+import akka.http.scaladsl.model._
 import de.heikoseeberger.akkahttpcirce.ErrorAccumulatingCirceSupport._
 import io.circe.generic.auto._
 import io.buildo.enumero.circe._
 
+import Directives._
+import StatusCodes._
+import model._
 import Game._
-import model.{Move, Result, RequestBody}
 import Result._
 
 object Main extends App {
   implicit val system = ActorSystem("my-system")
   implicit val materializer = ActorMaterializer()
   implicit val executionContext = system.dispatcher
+  implicit def myRejectionHandler =
+    RejectionHandler
+      .newBuilder()
+      .handleAll[MalformedRequestContentRejection] { rejection =>
+        complete(WrongMove())
```

Comment:
Have you tried using `reject.message` to provide a nicer error?

Reply:
Also, this completes with 200 I think. You want a 400 in this case

Reply:
I switched to `422` as response status but I don't see any message on akka's `reject`...